packages/block-editor/src/store/private-actions.js

```diff
@@ -30,11 +30,13 @@ const privateSettings = [
  *
  * @param {Object}  settings                  Updated settings
  * @param {boolean} stripExperimentalSettings Whether to strip experimental settings.
+ * @param {boolean} reset                     Whether to reset the settings.
  * @return {Object} Action object
  */
 export function __experimentalUpdateSettings(
 	settings,
-	stripExperimentalSettings = false
+	stripExperimentalSettings = false,
+	reset = false
```

Comment:
```suggestion
	{
		stripExperimentalSettings = false,
		reset = false
	}
```

**Nit**: these positional arguments are growing and require reading the source of the function in order to understand what a given call is doing.

For example what does `true` do here (I know the answer but only because I'm reviewing this PR)?

```js
__experimentalUpdateSettings(
	{
		...settings,
		__internalIsInitialized: true,
	},
	stripExperimentalSettings
	stripExperimentalSettings,
	true
);
```

Perhaps we can consider rewriting to use an object?

Reply:
Sure, it's a private action though, we can change our mind at any time.

Reply:
Doesn't need to happen now of course.